File: 01_GOF23/DP_examples/Python/02_Structural/Adapter/target.py
from abc import ABC, abstractmethod
from adaptee import MediaType


# 目标接口
class IPlayer(ABC):
    @abstractmethod
    def play(self, file_name, typ):
        pass

# 具体目标类
class Player(IPlayer):
    def play(self, file_name, typ):
        if typ == MediaType.MP3:
            print(f"PlayAudio mp3 : {file_name}")

File: 01_GOF23/DP_examples/Rust/02_Structural/Adapter/target.rs
use crate::adaptee::MediaType;

// 目标接口
pub trait IPlayer {
    fn play(&self, file_name: &str, typ: MediaType);
}

// 具体目标类
pub struct Player;

impl IPlayer for Player {
    fn play(&self, file_name: &str, typ: MediaType) {
        if typ == MediaType::MP3 {
            println!("PlayAudio mp3 : {}", file_name);
        }
    }
}